ContourTreeAlignment - New Example and Documentation

floWetzels opened the pull request:
I added a simplified example that only computes the alignment, because the old one used an extractSelection filter which is not compatible with python states. The new example exists as a python and a paraview state file, the old one only as a paraview state. There are now the following example states:
states/contourTreeAlignment.pvsm
python/contourTreeAlignment.py
states/contourTreeAlignmentWithMatching.pvsm
There is one documentation file for all three states.

Commit:
started work on alignment example

Commit:
Merge branch 'dev' of https://github.com/topology-tool-kit/ttk-data into dev

Commit:
finished alignment example

Commit:
Merge branch 'dev' of https://github.com/topology-tool-kit/ttk-data into dev

Commit:
BuiltInExample1 python state and docs

floWetzels commented:
Sorry, I forgot to change the branch. This PR now also contains the python state and docs for BuiltInExample1.

julien-tierny commented on python/contourTreeAlignment.py:
```diff
@@ -0,0 +1,51 @@
+# create a new 'TTK Merge and Contour Tree (FTM)'
+ttkMergeandContourTreeFTM = TTKMergeandContourTreeFTM(registrationName='TTKMergeandContourTreeFTM', Input=ttkTopologicalSimplificationByPersistence)
+ttkMergeandContourTreeFTM.ScalarField = ['POINTS', 'nrrd']
+ttkMergeandContourTreeFTM.InputOffsetField = ['POINTS', 'nrrd']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/contourTreeAlignment.py:
```diff
@@ -0,0 +1,51 @@
+alignmentEdges.CoordinateResults = 1
+alignmentEdges.Function = 'iHat*Layout_Y+jHat*Scalar*3'
+
+# create a new 'Calculator'
```
pipeline dead leaf, please remove.

julien-tierny commented on python/contourTreeAlignment.py:
```diff
@@ -0,0 +1,51 @@
+alignmentEdges.Function = 'iHat*Layout_Y+jHat*Scalar*3'
+
+# create a new 'Calculator'
+alignmentNodes = Calculator(Input=alignmentEdges)
```
dead leaf

julien-tierny commented on python/contourTreeAlignment.py:
```diff
@@ -0,0 +1,51 @@
+
+# create a new 'Calculator'
+alignmentNodes = Calculator(Input=alignmentEdges)
+alignmentNodes.Function = ''
```
dead leaf

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+
+# create a new 'XML Image Data Reader'
+builtInExamplevti = XMLImageDataReader(FileName=['BuiltInExample1.vti'])
+builtInExamplevti.PointArrayStatus = ['Vectors_']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+
+# create a new 'Compute Derivatives'
+computeDerivatives1 = ComputeDerivatives(Input=transform1)
+computeDerivatives1.Scalars = [None, '']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+# create a new 'TTK PersistenceDiagram'
+tTKPersistenceDiagram1 = TTKPersistenceDiagram(Input=tetrahedralize1)
+tTKPersistenceDiagram1.ScalarField = ['POINTS', 'myVorticity']
+tTKPersistenceDiagram1.InputOffsetField = [None, '']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+tTKPersistenceDiagram1.ScalarField = ['POINTS', 'myVorticity']
+tTKPersistenceDiagram1.InputOffsetField = [None, '']
+
+# create a new 'Threshold'
```
dead leaf, please remove.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+diagonal.ThresholdRange = [-1.0, -0.1]
+
+# create a new 'Extract Surface'
+extractSurface1 = ExtractSurface(Input=diagonal)
```
dead leaf, please remove

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+extractSurface1 = ExtractSurface(Input=diagonal)
+
+# create a new 'Tube'
+tube1 = Tube(Input=extractSurface1)
```
dead leaf, please remove

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+maxima.ThresholdRange = [3.0, 3.0]
+
+# save the output
+SaveData('minima.vtu', minima)
```
for a batch mode usage, I'd recommend to store the raw output data instead of graphical primivites.
in this example, points instead of spheres.
this is more convenient and generic for possible post-processing.

you may want to use `tTKScalarFieldCriticalPoints1` directly (or subsets, if so, you may need to use `MaskPoints` prior to thresholding).
also, please store that in `csv` file format, which is more convenient for a batch mode python noobie

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+
+# save the output
+SaveData('minima.vtu', minima)
+SaveData('maxmima.vtu', maxima)
```
same as above

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+extractSurface2 = ExtractSurface(Input=persistenceThreshold)
+
+# create a new 'Tube'
+tube2 = Tube(Input=extractSurface2)
```
dead leaf, please remove.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+# create a new 'TTK PersistenceCurve'
+tTKPersistenceCurve1 = TTKPersistenceCurve(Input=tetrahedralize1)
+tTKPersistenceCurve1.ScalarField = ['POINTS', 'myVorticity']
+tTKPersistenceCurve1.InputOffsetField = ['POINTS', '']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+tTKTopologicalSimplification1 = TTKTopologicalSimplification(Domain=tetrahedralize1,
+    Constraints=tTKIcospheresFromPoints1)
+tTKTopologicalSimplification1.ScalarField = ['POINTS', 'myVorticity']
+tTKTopologicalSimplification1.InputOffsetField = [None, '']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+    Constraints=tTKIcospheresFromPoints1)
+tTKTopologicalSimplification1.ScalarField = ['POINTS', 'myVorticity']
+tTKTopologicalSimplification1.InputOffsetField = [None, '']
+tTKTopologicalSimplification1.VertexIdentifierField = [None, '']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+warpByScalar1 = WarpByScalar(Input=tTKTopologicalSimplification1)
+warpByScalar1.Scalars = ['POINTS', 'myVorticity']
+warpByScalar1.ScaleFactor = 300.0
+warpByScalar1.UseNormal = 1
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny commented on python/BuiltInExample1.py:
```diff
@@ -0,0 +1,121 @@
+# create a new 'TTK ScalarFieldCriticalPoints'
+tTKScalarFieldCriticalPoints1 = TTKScalarFieldCriticalPoints(Input=warpByScalar1)
+tTKScalarFieldCriticalPoints1.ScalarField = ['POINTS', 'myVorticity']
+tTKScalarFieldCriticalPoints1.InputOffsetField = ['POINTS', 'OutputOffsetScalarField']
```
I don't believe this line is necessary.
Thanks for double checking (e.g. with diff or paraview) that the outputs of the script are unchanged after the edit.

julien-tierny reviewed (changes requested):
thanks.
please move `states/contourTreeAlignmentWithMatching.pvsm` to a distinct PR.
next time, thanks for doing only one PR by example.

julien-tierny commented:
once your PR is merged, thanks for including a link to this example in the doxygen documentation of the involved modules.
see this commit for instance:
topology-tool-kit/ttk@2b9fd4c

Commit:
removed unnecessary lines

Commit:
removed dead ends

Commit:
Changed output from minima/maxima spheres to simply critical points

Commit:
updated documentation for shortened python files

Commit:
removed new state file again, modified python statefile such that it works with ExtractSelection and now show full pipeline with matching

Commit:
updated documentation for statefile changes

Commit:
changed output format of critical points to csv and updated documentaiton

floWetzels commented:
I updated the built in example with the requested changes.
For the ContourTreeAlignment example, I went back to just one state file as discussed, added the vertex selection and updated the docs accordingly.

julien-tierny committed:
Slight updates to BuiltInExample1

julien-tierny committed:
slight edits to contourTreeAlignment

julien-tierny commented:
rock'n'roll, thanks a lot!

julien-tierny commented:
One more thing, can you please update the doxygen documentation of the VTK layer classes involved in the examples of this PR, in order to include a link to the example? (and open a PR to TTK's code respository)

For that, please double-check this commit for an example:
https://github.com/topology-tool-kit/ttk/commit/2b9fd4c

Thanks!
